util/package.go

```diff
@@ -185,8 +191,18 @@ func NewPackage(basePath string) (*Package, error) {
 
 	p.Downloads = []Download{NewDownload(*p, "tar")}
 
-	if p.Requirement.Kibana.Versions != "" {
-		p.Requirement.Kibana.semVerRange, err = semver.NewConstraint(p.Requirement.Kibana.Versions)
+	// If the new conditions are used, select them over the requirements
```

Comment:
There is duplicated code in this if-else to account for the new and legacy settings. Looks like the intent is to use the new setting, `p.Conditions.KibanaVersion`. So why not do something like this instead?

```
p.setKibanaVersion()
if p.Conditions != nil && p.Conditions.KibanaVersion != "" {
   ...
}
```

```
func (p *Package) setKibanaVersion() {
  if p.Conditions != nil && p.Conditions.KibanaVersion != "" {
    return
  }

  if p.Requirement.Kibana.Versions != "" {
    if p.Conditions == nil {
      // initialize p.Conditions
    }
    p.Conditions.KibanaVersion = p.Requirement.Kibana.Versions
  }
}    
```

Reply:
@ycombinator The legacy code should only be around for 24-48h, so I was sloppy.